pkg/machines/libvirt-dbus.js

```diff
@@ -1312,4 +1313,9 @@ export function storageVolumeDelete(connectionName, poolName, volName) {
             .then(objPath => call(connectionName, objPath[0], 'org.libvirt.StorageVol', 'Delete', [0], TIMEOUT));
 }
 
+export function vmInterfaceAddresses(connectionName, objPath) {
+    // 'lease' source will work only for those domains which take IP address from libvirt spawned dnsmasq (i.e. have <interface type='network'/>)
+    return call(connectionName, objPath, 'org.libvirt.Domain', 'InterfaceAddresses', [Enum.VIR_DOMAIN_INTERFACE_ADDRESSES_SRC_LEASE, 0], TIMEOUT);
```

Comment:
Just an idea. But calling this API with LEASE can find only IP adresses which have been assigned to interface by libvirt itself. But there are cases where IP address have been assigned to interface "from outside by outside dns". For example when interface is of type direct. Then libvirt does not assign IP to this interface, and LEASE won't return those IP addresses.
So perhaps we could do both calls (LEASE, and if LEASE won't find IP address, then try AGENT). Altought as you mentioned earlier on IRC, you are right we cannot rely on presence of qemu guest agent. Then perhaps we could try libvirt nss module, which should return all IP adresses?

**But this is just an idea how to "enlarge" this functionality**. It could be done in another PR. Most user cases where user wants to know interface's IP address are covered by API call with LEASE source.

Reply:
What does it mean to 'Then perhaps we could try libvirt nss module, which should return all IP adresses?'

Reply:
Sorry, I didn't realize nss module doesn't affect some API which we could use here. So disregard what I said about nss module.
But yeah, we could add fallback to guest agent info in the future. It could find some IP adresses which API call with lease could not.

Reply:
I've tried to persuade Simon to have cockpit install & configure NSS plugin. The reasoning behind is to make user experience smoother. I mean, I might be missing what this feature is about, but one of the use cases to me is to get an IP address to ssh to. Well, for that I need to open browser, do bunch of clicks to find out an IP address then switch back to console and paste it.
Whereas with NSS plugin I can just 'ssh root@virtualMachine' right away: [https://libvirt.org/nss.html](docs)

Reply:
@zippy2 Good point. I guess it would not hurt to add it in Recommended packages in the spec file. However we do have bultin terminal in the browser for the VMs so I wonder why someone would want to ssh there.